src/vm/castcache.h

```diff
@@ -0,0 +1,392 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// File: castcache.h
+//
+
+#ifndef _CAST_CACHE_H
+#define _CAST_CACHE_H
+
+#include "util.hpp"
+#include "syncclean.hpp"
+
+//
+// A very lightweight cache that maps {source, target} -> result, where result is 
+// a boolean value indicating that the types are definitely convertible or definitely not.
+// We generically allow either MethodTable* or TypeHandle as source/target. Either value
+// uniquely maps to a single type with no possibility of confusion. Besides for most types 
+// TypeHandle a MethodTable* are the same value anyways.
+//
+// The primary purpose is caching results of conversion analysis assuming that covertibility relationship, 
```

Comment:
This is using C# spec terminology. There is discrepancy between what conversions are in C# language spec and what conversions are in ECMA-335 spec.

We stick to to ECMA-335 terminology in the runtime. I think this should talk about casting.

Reply:
I did mean "in the ECMA-335" sense. I.E. what spec calls "compatible-with". - Basically the relation used by `castclass` and `isinst`.

I will make the comments less confusing by removing "conversion" as a term. It can be ambiguous.

---
In reply to: [306636144](https://github.com/dotnet/coreclr/pull/23548#discussion_r306636144) [](ancestors = 306636144)